### DISCUSSION TITLE
[Idea] $ as shell global

### DISCUSSION THREAD

**Discussion by DavidWells** [johnlindquist/kit#258] [Ideas]
Found this interesting project today that might make a good addition to global API

https://github.com/google/zx

```js
await $`cat package.json | grep name`

let branch = await $`git branch --show-current`
await $`dep deploy --branch=${branch}`

await Promise.all([
  $`sleep 1; echo 1`,
  $`sleep 2; echo 2`,
  $`sleep 3; echo 3`,
])

let name = 'foo bar'
await $`mkdir /tmp/${name}`
```

Just a thought ❤️

**Comment by johnlindquist** [comment]
Kit exposes most of shelljs as globals. 

https://github.com/shelljs/shelljs

My preference is small, composable functions. What's the appeal of a $ global where the functions are inside a string?

**Reply by craigmichaelmartin** [reply]
Interesting! I like proposal. The appeal to me is not needing to learn an intermediary API to express the thing I already know how to express.

**Reply by johnlindquist** [reply]
I think that's an excellent point: "Let me do what I already know"

**Comment by johnlindquist** [comment]
`$` is now supported in beta.17: https://github.com/johnlindquist/kit/discussions/282
